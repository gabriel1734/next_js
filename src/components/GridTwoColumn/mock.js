export default {
  title: 'Grid Two Column',
  text: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Laboriosam, aut labore! Dolores cupiditate reprehenderit, fugit ab provident ad sequi nisi iure. Tenetur at cupiditate atque, tempore nihil eum repellendus provident?`,
  srcImg: 'assets/images/javascript.svg',
};
