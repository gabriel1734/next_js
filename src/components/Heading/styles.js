import styled, { css } from 'styled-components';

const titleSize = {
  small: (theme) => css`
    font-size: ${theme.fonts.sizes.medium};
  `,
  medium: (theme) => css`
    font-size: ${theme.fonts.sizes.large};
  `,
  big: (theme) => css`
    font-size: ${theme.fonts.sizes.xlarge};
  `,
  huge: (theme) => css`
    font-size: ${theme.fonts.sizes.xhuge};
    ${mediaFont(theme)};
  `,
};

const mediaFont = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.fonts.sizes.xlarge};
    background-color: 'red';
  }
`;

export const Title = styled.h1`
  ${({ theme, colorDark, size, uppercase }) => css`
    color: ${colorDark ? theme.colors.white : theme.colors.primaryColor};
    ${titleSize[size](theme)};
    text-transform: ${uppercase ? 'uppercase' : 'none'};
  `}`;
