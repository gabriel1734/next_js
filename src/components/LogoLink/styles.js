import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    color: ${theme.colors.primaryColor};
    text-decoration: none;
    

    > img {
      height: 3rem;
    }
  `}
`;
