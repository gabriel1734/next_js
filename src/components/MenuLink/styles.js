import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    display: block;
    text-decoration: none;
    font-size: ${theme.fonts.sizes.medium};
    padding: ${theme.spacings.small};
    color: ${theme.colors.primaryColor};
    position: relative;
    

    &::after {
      content: '';
      position: absolute;
      bottom: 0.76rem;
      left: 50%;
      width: 0%;
      height: 0.2rem;
      background-color: ${theme.colors.secondaryColor};
      transition: all 0.5s ease-in-out;
    }
    &:hover::after {
      left: 25%;
      width: 50%;
    }
  `}
`;
