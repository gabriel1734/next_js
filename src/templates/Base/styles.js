import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${() => css`
    padding-top: 2rem;
  `}
`;
