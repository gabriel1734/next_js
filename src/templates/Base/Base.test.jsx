import { container } from 'webpack';
import { Base } from '.';
import { renderTheme } from '../../styles/render-theme';
import mock from './mock';

describe('<Base />', () => {
  it('should render', () => {
    renderTheme(<Base {...mock} />);
    expect(container).toMatchSnapshot();
  });
});
