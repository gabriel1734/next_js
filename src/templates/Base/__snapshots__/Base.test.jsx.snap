// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Base /> should render 1`] = `
Object {
  "ContainerPlugin": [Function],
  "ContainerReferencePlugin": [Function],
  "ModuleFederationPlugin": [Function],
  "scope": [Function],
}
`;
